Fix getTextContent evaluation to only apply TJ horizontal offsets using numeric items/args

cemerick opened the pull request:
While the array argument to TJ should only contain strings and numbers, other
unfortunate items are found in PDFs in the wild, e.g.:

```
[(Grandes) 0.0 Tc
-250.0 (Client\350les,) 0.0 Tc
-250.0 (Financements) 0.0 Tc
-250.0 (et) 0.0 Tc
-250.0 (March\351s) ] TJ
```

getOperatorList already properly ignores any non-string, non-numeric values in
TJ arrays; without this patch to getTextContent, returned text items can have
NaN widths due to calculations being applied to those non-numeric values.

yurydelendik commented:
Can you provide some form of test (e.g. ref text)?

cemerick commented:
You mean a sample PDF that contains an out-of-spec TJ array? Not at the moment, the example I have is from a customer.

FWIW, the change is just mirroring what getOperatorList does already.

yurydelendik commented:
Hmm, we might have 'eq' test for it -- it's probably a matter of adding 'text' one

cemerick committed:
Fix getTextContent evaluation to only apply TJ horizontal offsets using numeric items/args

While the array argument to TJ should only contain strings and numbers, other
unfortunate items are found in PDFs in the wild, e.g.:

[(Grandes) 0.0 Tc
-250.0 (Client\350les,) 0.0 Tc
-250.0 (Financements) 0.0 Tc
-250.0 (et) 0.0 Tc
-250.0 (March\351s) ] TJ

getOperatorList already properly ignores any non-string, non-numeric values in
TJ arrays; without this patch to getTextContent, returned text items can have
NaN widths due to calculations being applied to those non-numeric values.

cemerick commented:
Updated with a minimized test case. Hopefully I got the manifest entry right.

FWIW, here's an easy expression to evaluate once the testcase PDF is loaded in the viewer:

```js
PDFViewerApplication.pdfDocument.getPage(1).then((p) => p.getTextContent()).then((tc) => console.log(tc.items[0]));
```

You'll see the sole textItem; its width will be `NaN` using HEAD, ~193 with the patch. Turns out it is affecting the text layer (positively): the `scaleX` css property is being calculated properly, so the text run's div now covers the rendered text fully.

timvandermeij commented on test/pdfs/.gitignore:
```diff
@@ -258,3 +258,5 @@
 !annotation-choice-widget.pdf
 !zero_descent.pdf
+!operator-in-TJ-array.pdf
+
```
Nit: please remove this newline

cemerick replied:
Done.

timvandermeij commented:
/botio-linux preview

timvandermeij commented:
/botio test

timvandermeij commented:
/botio makeref

timvandermeij commented:
Thank you for your contribution!
